src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -81,26 +122,103 @@ export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
 
   const onEventBoxClick = useCallback(
     (e) => {
-      if (e.target.closest(`.${ScheduleEventBookmarkClass}`)) return;
+      if (
+        e.target.closest(`.${ScheduleEventBookmarkClass}`) ||
+        e.target.closest(`.${ScheduleEventExpandClass}`)
+      )
+        return;
 
       showEventModal();
     },
     [showEventModal]
   );
 
+  const handleMouseOver = useCallback(() => {
+    if (isMinimallyLongEvent) return;
+
+    showExpanded();
+  }, [isMinimallyLongEvent, showExpanded]);
+
+  const handleMouseOut = useCallback(() => {
+    if (isExpandedStatic) return;
+
+    hideExpanded();
+  }, [isExpandedStatic, hideExpanded]);
+
+  const handleCollapse = useCallback(() => {
+    hideExpandedStatic();
+    hideExpanded();
+  }, [hideExpanded, hideExpandedStatic]);
+
+  const handleToggleExpand = useCallback(() => {
+    if (!isExpandedStatic) {
+      toggleExpandedStatic();
+
+      return;
+    }
+
+    handleCollapse();
+  }, [isExpandedStatic, handleCollapse, toggleExpandedStatic]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: Event) => {
+      const eventTarget = event.target as Node;
+      if (
+        eventRef.current &&
+        !eventRef?.current?.contains(eventTarget) &&
+        !isEventModalVisible
+      ) {
+        handleCollapse();
+      }
+    };
+
+    window.addEventListener("mousedown", handleClickOutside, false);
+
+    return () => {
+      window.removeEventListener("mousedown", handleClickOutside, false);
+    };
+  }, [eventRef, handleCollapse, isEventModalVisible]);
+
   return (
     <>
-      <div className={containerClasses} onClick={onEventBoxClick}>
-        <div className="ScheduleEvent__info">
-          <div className="ScheduleEvent__title">{event.name}</div>
-          <div className="ScheduleEvent__host">by {event.host}</div>
-        </div>
-
-        <div className={ScheduleEventBookmarkClass} onClick={bookmarkEvent}>
-          <FontAwesomeIcon
-            icon={event.isSaved ? solidBookmark : regularBookmark}
-          />
-        </div>
+      <div
+        ref={eventRef}
+        className={containerClasses}
+        onClick={onEventBoxClick}
+        onMouseEnter={handleMouseOver}
+        onMouseLeave={handleMouseOut}
+      >
+        {isShowExpand && (
+          <div
+            className={ScheduleEventExpandClass}
+            onClick={handleToggleExpand}
+          >
```

Comment:
Using a `div` with `onClick` like this creates accessibility issues. Any reason why we can't use `button` here?

Reply:
done f718762